### PR TITLE
Add Time to Date Range Picker in Patient Filter Page

### LINKED ISSUE #1076: Filter by time : advanced filter : patient list
Please add the time option with the Date range option on the advanced filter on the patient list.

Eg: in Created after/ created before/modified after- time 

The backend is already done.

### PR DESCRIPTION
**Pull request by abhinandanarya06**
## Proposed Changes
1. Added `src/Components/Common/DateTimeRangePicker.tsx` which is custom material ui based date time range picker. 
    - **Reason**: Couldn't use another because other don't have corresponding `@types` available and gives many errors with jsx.
2. Updated Patient Filter page to use above component 

## Merge Checklist
- [ ] Filter correctly working
- [ ] UI/UX Improved

bdfd9f95-3074-4929-b752-6b4f404f2042

closes #1076

### CODE REVIEW

File: src/Components/Common/DateTimeRangePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+export const DateTimeRangePicker: React.FC<IDateTimeRangePickerProps> = ({
+  label,
+  endDateId = "end_date",
+  endDate,
+  startDate,
+  onChange,
+  startDateId = "start_date",
+  size = "regular",
+}) => {
+  const [focusInput, setFocusInput] = React.useState<any>(null);
```
Comment (developedBySJ) [inline src/Components/Common/DateTimeRangePicker.tsx]: Remove this line as it is not needed

Comment (ghost) [reply src/Components/Common/DateTimeRangePicker.tsx]: oh forgot that thanks

**Review by developedBySJ — COMMENTED** [review]
Looks good

File: src/Components/Common/DateTimeRangePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      <div className="flex">
+        {label && <span className="text-sm font-semibold">{label}</span>}
+        <button
+          className="text-xs font-semibold px-2 bg-red-200 hover:bg-red-300 rounded uppercase float-right mr-1 ml-auto"
+          onClick={() => {
+            onChange({ startDate: null, endDate: null });
+          }}
+        >
+          X Clear
+        </button>
```
Comment (khavinshankar) [inline src/Components/Common/DateTimeRangePicker.tsx]:
```suggestion
        {(startDate || endDate) && <button
          className="text-xs font-semibold px-2 bg-red-200 hover:bg-red-300 rounded uppercase float-right mr-1 ml-auto"
          onClick={() => {
            onChange({ startDate: null, endDate: null });
          }}
        >
          X Clear
        </button>}
```

**Review by khavinshankar — APPROVED** [review]
```js
{(startDate || endDate) && <button
          className="text-xs font-semibold px-2 bg-red-200 hover:bg-red-300 rounded uppercase float-right mr-1 ml-auto"
          onClick={() => {
            onChange({ startDate: null, endDate: null });
          }}
        >
          X Clear
        </button>}
```


add a simple check so that the `X Clear` shows only when there is some value.

rest LGTM

File: src/Components/Common/DateTimeRangePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    endDate: moment.Moment | null;
+  }) => void;
+  startDate: moment.Moment | null;
+  endDate: moment.Moment | null;
+  startDateId?: string;
+  endDateId?: string;
+  size?: "regular" | "small";
+}
+
+export const getDate = (value: any) =>
```
Comment (smartclash) [inline src/Components/Common/DateTimeRangePicker.tsx:20]: let's not use `any`

**Review by smartclash — CHANGES_REQUESTED** [review]

**Comment by bodhish** [comment]
Let's not merge this now. The other picker was way better for data selection. 

@aparnacoronasafe can you re-evaluate the actual use of date-time again?

**Comment by nihal467** [comment]
@gigincg its been blocked for a long time now, shall we close it

**Comment by gigincg** [comment]
Stale/Archived

**Pull request closed without merging**